packages/module/patternfly-docs/content/extensions/data-view/examples/Toolbar/Toolbar.md

```diff
@@ -0,0 +1,143 @@
+
+**Initial values:**
+- `perPage` initial value
+- Optional `page` initial value
+- Optional `searchParams` object
+- Optional `setSearchParams` function
+
+While the hook works seamlessly with the React Router library, you do not need to use it to take advantage of URL persistence. The `searchParams` and `setSearchParams` props can be managed using native browser APIs (`URLSearchParams` and `window.history.pushState`) or any other routing library of your choice. If you don't pass these two props, the pagination state will be stored internally without the URL usage.
+
+You can also pass custom `pageParam` or `perPageParam` names, renaming the pagination parameters in the URL.
+
+The retrieved values are named to match the PatternFly pagination component props.
+
+**Return values:**
+- Current `page` number
+- `onSetPage` to modify current page
+- Items `perPage` value
+- `onPerPageSelect` to modify per page value
+
+### Pagination example
+This example uses the URL to persist the pagination state.
+
+```js file="./PaginationExample.tsx"
+
+```
+
+## Selection
+To allow users to select data records inside the data view, add selection support that displays a row's selection state.
+
+The data view toolbar can display a bulk selection component using bulkSelect. You can make use of a predefined [component group extension bulk select](/extensions/component-groups/bulk-select) component.
+
+### Selection state
+
+The `useDataViewSelection` hook manages the selection state of the data view. 
+
+**Initial values:**
+- Optional `initialSelected` array of record's identifiers selected by default 
+- `matchOption` function to check if given record is selected
+
+*When no `matchOption` is passed, the `Array.prototype.includes()` operation is performed on the `selected` array.*
+
+**Return values:**
+- `selected` array of currently selected records
+- `isSelected` function returning the selection state for given record
+- `onSelect` callback to modify the selection state and accepting `isSelecting` flag indicating if records are changing to selected or deselected and `items` containing affected records 
+
+### Selection example
+
+```js file="./SelectionExample.tsx"
+
+```
+
+## Filters
+To allow users to filter data records in the data view, add filtering support that displays the applied filter chips.
+
+The data view toolbar can include a set of filters by passing a React node to the `filters` property. You can use the predefined components `<DataViewFilters>`, `<DataViewTextFilter>`, and `<DataViewCheckboxFilter>` to customize and handle filtering directly in the toolbar. The `<DataViewFilters>` component is a wrapper that allows conditional filtering using multiple attributes. If you need just a single filter, you can use `<DataViewTextFilter>`, `<DataViewCheckboxFilter>`, or a different filter component alone. Props of these filter components are listed at the bottom of this page. 
```

Comment:
```suggestion
The data view toolbar can include a set of filters by passing a React node to the `filters` property. You can use the predefined components `<DataViewFilters>`, `<DataViewTextFilter>`, and `<DataViewCheckboxFilter>` to customize and handle filtering directly in the toolbar. The `<DataViewFilters>` component is a wrapper that allows conditional filtering using multiple attributes. If you need just a single filter, you can use `<DataViewTextFilter>`, `<DataViewCheckboxFilter>`, or a different filter component alone. Props of these filter components are listed in the [props section of this page](#props).
```
we just typically avoid words like below/above for accessibility purposes. so however you'd like to adjust that!